apps/estuary/lib/estuary/message_handler.ex

```diff
@@ -1,9 +1,33 @@
 defmodule Estuary.MessageHandler do
+  @moduledoc """
+  Estuary.MessageHandler reads an event from the event stream and persists it.
+  """
+  alias Estuary.EventTable
   use Elsa.Consumer.MessageHandler
   require Logger
 
   def handle_messages(messages) do
-    Logger.debug("Messages #{inspect(messages)} were sent to the eventstream")
+    Enum.each(messages, fn message -> process_message(message) end)
+
+    Logger.debug("Messages #{inspect(messages)} were sent to the event stream")
     :ack
   end
+
+  defp process_message(message) do
+    case Jason.decode(message.value) do
+      {:ok, %{"author" => _, "create_ts" => _, "data" => _, "type" => _} = event} ->
+        EventTable.insert_event_to_table(event)
+
+      {:ok, term_without_keys} ->
+        Yeet.process_dead_letter("", message, "estuary",
+          reason:
+            "event #{inspect(term_without_keys)} was decoded but did not have the right keys"
+        )
+
+      {:error, %Jason.DecodeError{}} ->
```

Comment:
I'm not sure that we really need to differentiate between the above error and this one.

Reply:
The option above it was an ok tuple and it was handling the case where json can be decoded, but it didn't have the correct keys so it couldn't be inserted into the DB.

Reply:
It is still in the code, but I've broken it up into separate functions with different function signatures.

Reply:
My guess is that Jarred understands _how_ this code works and is stating he's not sure it's beneficial to handle those error cases individually instead of doing the same thing in either case. (You lose the granular error message, but how much does it buy us?)

```
case Jason.decode(message.value) do
      {:ok, %{"author" => _, "create_ts" => _, "data" => _, "type" => _} = event} ->
        EventTable.insert_event_to_table(event)

      _anything_else -> #yeet record
```